stdlib/Profile/src/Profile.jl

```diff
@@ -349,15 +349,18 @@ function getdict(data::Vector{UInt})
 end
 
 function getdict!(dict::LineInfoDict, data::Vector{UInt})
-    for ip in data
-        # Lookup is expensive, so do it only once per ip.
-        haskey(dict, UInt64(ip)) && continue
-        st = lookup(convert(Ptr{Cvoid}, ip))
-        # To correct line numbers for moving code, put it in the form expected by
-        # Base.update_stackframes_callback[]
-        stn = map(x->(x, 1), st)
-        try Base.invokelatest(Base.update_stackframes_callback[], stn) catch end
-        dict[UInt64(ip)] = map(first, stn)
+    # we don't want metadata here as we're just looking up ips
+    unique_data_itr = Iterators.unique(has_meta(data) ? strip_meta(data) : data)
```

Comment:
Since `unique` uses hashing which is much of what Dicts do, why is this a speed improvement?

Reply:
That just facilitates identifying the unique ips before going off to parallelize. It's the parallel part that helps most